contracts/external/cover/CoverBroker.sol

```diff
@@ -0,0 +1,190 @@
+  /// @param erc20 The ERC20 token for which to approve spending.
+  function maxApproveCoverContract(IERC20 erc20) external onlyOwner {
+    erc20.safeApprove(address(cover), type(uint256).max);
+  }
+
+  /// @notice Switches CoverBroker's membership to a new address.
+  /// @dev MemberRoles contract needs to be approved to transfer NXM tokens to new membership address.
+  /// @param newAddress The address to which the membership will be switched.
+  function switchMembership(address newAddress) external onlyOwner {
+    nxmToken.approve(address(memberRoles), type(uint256).max);
+    memberRoles.switchMembership(newAddress);
+  }
+
+  /// @notice Transfers all available funds of a specified asset (ETH or ERC20) to the contract owner.
+  /// @param assetAddress The address of the asset to be transferred.
+  function transferFunds(address assetAddress) external onlyOwner {
+
+    if (assetAddress == ETH) {
+      uint ethBalance = address(this).balance;
+      if (ethBalance == 0) {
+        revert ZeroBalance(ETH);
+      }
+
+      (bool sent, ) = payable(msg.sender).call{value: ethBalance}("");
+      if (!sent) {
+        revert TransferFailed(msg.sender, ethBalance, ETH);
+      }
+
+      return;
+    }
+
+    IERC20 asset = IERC20(assetAddress);
+    uint erc20Balance = asset.balanceOf(address(this));
+    if (erc20Balance == 0) {
+      revert ZeroBalance(assetAddress);
+    }
+
+    asset.transfer(msg.sender, erc20Balance);
+  }
+
+  /// @notice Handles ETH payments for buying cover.
+  /// @dev Calculates ETH refunds if applicable and sends back to msg.sender.
+  /// @param params The parameters required to buy cover.
+  /// @param poolAllocationRequests The allocation requests for the pool's liquidity.
+  /// @return coverId The ID of the purchased cover.
+  function _handleEthPayment(
+    BuyCoverParams calldata params,
+    PoolAllocationRequest[] calldata poolAllocationRequests
+  ) internal returns (uint coverId) {
+
+    uint ethBalanceBefore = address(this).balance - msg.value;
+    coverId = cover.buyCover{value: msg.value}(params, poolAllocationRequests);
+    uint ethBalanceAfter = address(this).balance;
+
+    // transfer any ETH refund back to msg.sender
+    if (ethBalanceAfter > ethBalanceBefore) {
+      uint ethRefund = ethBalanceAfter - ethBalanceBefore;
+      (bool sent, ) = payable(msg.sender).call{value: ethRefund}("");
+      if (!sent) {
+        revert TransferFailed(msg.sender, ethRefund, ETH);
+      }
+    }
+  }
+  
+  /// @notice Handles NXM payments for buying cover.
+  /// @dev Transfers NXM from the caller to the contract, then buys cover on behalf of the caller.
+  /// Calculates NXM refunds if any and sends back to msg.sender.
+  /// @param params The parameters required to buy cover.
+  /// @param poolAllocationRequests The allocation requests for the pool's liquidity.
+  /// @return coverId The ID of the purchased cover.
+  function _handleNxmPayment(
```

Comment:
_handleNxmPayment and _handleErc20Payment logic is very similar. if not for SafeERC20 we could have probably had the same code to handle both flows

Reply:
Yes, the differences I see:
1. you have to fetch the asset address in case of non-nxm
2. the approval target is the cover contract for non-nxm, and token controller for nxm

Perhaps we could fetch/figure the token address in the calling function and:
- either pass the target approval address and token address via params
- or set the approval directly and only pass the token address

Reply:
PS: Leaving them separate is also fine, I don't necessarily see an issue, especially because the nxm approval is done in the constructor.

Reply:
PPS: Rox was considering removing NXM support altogether, no strong opinions.

Reply:
i'll drop the NXM support as it'll make the contract simpler. I'll just add comments indicating that NXM payment for members should be done directly through Cover.buyCover